File: a_templates/floyd-warshall.cpp
#include <bits/stdc++.h>
using namespace std;
using ll = long long;

const int N = 2e5 + 5;

vector<array<int, 2>> g[N];

int main() {
    int n, m;
    cin >> n >> m;
    int s;
    cin >> s;
    vector<vector<ll>> dist(n + 1, vector<ll>(n + 1, 1e18));
    while (m--) {
        ll a, b, c;
        cin >> a >> b >> c;
        dist[a][b] = min(dist[a][b], c);
    }
    for (int k = 1; k <= n; k++) {
        for (int i = 1; i <= n; i++) {
            for (int j = 1; j <= n; j++) {
                dist[i][j] = min(dist[i][j], dist[i][k] + dist[k][j]);
            }
        }
    }
}

File: a_templates/floyd_warshall.py
n, m = map(int, input().split()) 
dist = [[float('inf') for _ in range(n+1)] for _ in range(n+1)] 

for i in range(n):
    dist[i][i] = 0

for _ in range(m):
    u, v, w = map(int, input().split())
    dist[u][v] = w

res = 0
for k in range(1, n+1):  # Intermediate nodes
    for i in range(1, n+1):  # Starting node
        for j in range(1, n+1):  # Ending node
            dist[i][j] = min(dist[i][j], dist[i][k] + dist[k][j])
            if dist[i][j] != float('inf'):
                res += dist[i][j]

print(res)